### DISCUSSION TITLE
unable Detection USRP  in Virtual Machine for srsRAN Project

### DISCUSSION THREAD

**Discussion by rajugbv** [srsran/srsRAN_Project#544] [General Help]
Greetings,

I am in the process of setting up a UE machine using a virtual machine on one laptop and have installed gNB on another laptop. My goal is to replicate the srsRAN gNB with the srsUE project. However, I am encountering an issue where the virtual machine is unable to detect the USRP, even after running uhd_find_device, which results in a "no device connected" message.

I am reaching out to request help in resolving this issue. Your expertise and any suggestions would be greatly appreciated.

Thank you.

**Comment by pgawlowicz** [comment]
You will need to connect the USB port (in case of b210 usage) to your VM. However, we do not recommend running gnb on a virtual machine, as it will cause a lof of performance issues.
